foo_uie_console/main.cpp

```diff
@@ -305,9 +323,19 @@ class EdgeStyleMenuNode : public uie::menu_node_popup_t {
     std::vector<uie::simple_command_menu_node> m_nodes;
 };
 
+class AppendNewlineMenuNode : public uie::simple_command_menu_node {
+public:
+    AppendNewlineMenuNode(service_ptr_t<ConsoleWindow> window)
+        : simple_command_menu_node("Append newline", "", window->get_append_newline() ? uih::Menu::flag_checked : 0,
```

Comment:
I think we need a better name for the menu item (I'm not sure I'd know what 'Append newline' does from the name).

I think 'Hide trailing newline' would be clearer (though it would mean reversing the meaning of the flag everywhere...) Alternatively, 'Show trailing newline' perhaps?